src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellSearchResultsRenderer.cs

```diff
@@ -37,7 +38,21 @@ DataTemplate DefaultTemplate
 				return _defaultTemplate ?? (_defaultTemplate = new DataTemplate(() =>
 					{
 						var label = new Label();
-						label.SetBinding(Label.TextProperty, SearchHandler.DisplayMemberName ?? ".");
+
+						if (RuntimeFeature.IsShellSearchResultsRendererDisplayMemberNameSupported)
+						{
+							label.SetBinding(Label.TextProperty, SearchHandler.DisplayMemberName ?? ".");
+						}
+						else
+						{
+							if (SearchHandler.DisplayMemberName is not null)
+							{
+								Application.Current?.FindMauiContext()?.CreateLogger<ShellSearchResultsRenderer>().LogWarning(TrimmerConstants.SearchHandlerDisplayMemberNameNotSupportedWarning);
```

Comment:
Should this `throw`? and then we put `[RequiresUnreferencedCode]` on the setter of the `DisplayMemberName` property?

Reply:
This is exactly the code that I had locally, and it produced the flood of trimming warnings (we can't apply the attribute to fields BTW). Maybe the warnings are a bug in ILC? @vitek-karas might know what's going on when he's back from vacation.

Reply:
Ok, it's probably not a bug, but it's still strange. We get 2 exactly same warnings on each line where we use `typeof(SearchHandler)`, for example:
```c#
// /.../maui/src/Controls/src/Core/Shell/SearchHandler.cs(145): Trim analysis warning IL2026: Microsoft.Maui.Controls.SearchHandler..cctor(): Using member 'Microsoft.Maui.Controls.SearchHandler.DisplayMemberName.set' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. Use ItemTemplate instead. [/.../MyMauiApp/MyMauiApp.csproj::TargetFramework=net9.0-maccatalyst]
// /.../maui/src/Controls/src/Core/Shell/SearchHandler.cs(145): Trim analysis warning IL2026: Microsoft.Maui.Controls.SearchHandler..cctor(): Using member 'Microsoft.Maui.Controls.SearchHandler.DisplayMemberName.set' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. Use ItemTemplate instead. [/.../MyMauiApp/MyMauiApp.csproj::TargetFramework=net9.0-maccatalyst]
145: public static readonly BindableProperty CancelButtonColorProperty = BindableProperty.Create(nameof(CancelButtonColor), typeof(Color), typeof(SearchHandler), default(Color));
```

At the same time, the compiled XAML doesn't produce any warnings. That is because XamlC doesn't use the setter method, but it sets the value using `BindableObject.SetValue`:
```xml
<Shell.SearchHandler>
    <local:AnimalSearchHandler Placeholder="Enter search term"
                               ShowsResults="true"
                               DisplayMemberName="Name" />
</Shell.SearchHandler>
```
```c#
// decompiled IL
animalSearchHandler.SetValue(SearchHandler.DisplayMemberNameProperty, "Name");
```

Reply:
This is very likely a problem in the trimmer - and possibly even in NativeAOT, but that I'm not sure of. The root cause is that the parameter to which we're passing `typeof(SearchHandler)` has `DAM(PublicProperties | PublicMethods)`.

The `PublicProperties` will mark the `DisplayMemberName` property as reflection accessible, which in turn marks the getter of it as reflection accessible -> warning.
The `PublicMethods` will mark the getter direcly as reflection accessible -> warning.

@sbomer as FYI.